pandas/core/nanops.py

```diff
@@ -474,8 +474,7 @@ def nanargmax(values, axis=None, skipna=True):
     """
     Returns -1 in the NA case
     """
-    values, mask, dtype, _ = _get_values(values, skipna, fill_value_typ='-inf',
-                                         isfinite=True)
```

Comment:
Not sure what you mean by “it doesn't work now anyhow”? I think the build failures were just a linting error that I missed locally; trying again.

I think the argument for allowing it with strings is consistency: right now, the Pandas version of `argmax` etc is inconsistent with base Python, with NumPy, and with others Pandas functions. For example,

```python
data = ['spam', 'eggs']
data_a = np.array(data, dtype='object')
data_s = pd.Series(data)

max(data) # -> 'spam'
data_a.max() # -> 'spam'
data_s.max() # -> 'spam'

# So far so good

data.index(max(data)) # -> 0
data_a.argmax() # -> 0
data_s.argmax() #-> ValueError: could not convert string to float: 'eggs'
```

This is confusing; we expect data_s.argmax() to return 0 just like everything else. After this PR, it does return 0, as expected.

It's true that if we change the example above to use mixed data types, like `data2 = ['spam', 'eggs', None]`, then the behavior is different under Python 2 and Python 3, but I think that even in this case this PR makes the Pandas behavior more consistent with the prevailing behavior of whatever version of Python you're using. For example, 

```python
data2 = ['spam', 'eggs', None]
data2_a = np.array(data2, dtype='object')
data2_s = pd.Series(data2)

max(data2) # -> 'spam' under Py2, TypeError under Py3
data2_a.max() # -> 'spam' under Py2, TypeError under Py3
data2_s.max() # -> 'spam' under Py2, TypeError under Py3

data2.index(max(data2)) # -> 0 under Py2, TypeError under Py3
data2_a.argmax() # -> 0 under Py2, TypeError under Py3
```

Now, with the current release of Pandas, `data2_s.argmax()` will throw a `ValueError` under Py2 and Py3 with a message about converting strings to floats. After this PR, `data2_s.argmax()` will return 0 under Py2, as we would expect, and will throw a `TypeError` with a message about incomparable types under Py3.